### DISCUSSION TITLE
illegal hardware instruction/Apple M1: python -m spacy download

### DISCUSSION THREAD

**Discussion by superseoworld** [explosion/spaCy#8595] [Help: Installation]
first of all: spacy is great software.
i have a problem installing a model. calling python -m spacy download throws the error illegal harware instruction. what can i do? thanks a lot regards thomas

**Comment by polm** [comment]
Hey, sorry you're having trouble with this. We've had [one report](https://stackoverflow.com/questions/68076590/cant-install-ru-core-news-md-en-core-web-md-for-spacy-3-0-6-through-cli-on-app) of this problem, though the user didn't follow up so we haven't been able to confirm details. 

Can you give us the output of `python -m spacy info` or does that give an error too?

What version of Python are you using, and how did you install spaCy?

**Reply by superseoworld** [reply]
Hi, thanks for your reply. 

python -m spacy info produces the same error.

I have installed spacy as follows:

pip install -U spacy


Thanks. Regards Thomas

**Reply by polm** [reply]
Based on https://github.com/allenai/scispacy/issues/372 this might be an issue with nmslib. 

Can you `import nmslib`? 

You can try this to fix it:

```
pip uninstall nmslib
pip install --no-binary :all: nmslib
```

And then try `import nmslib` and using spaCy again.

**Reply by superseoworld** [reply]
Thanks.

nmslib is installable and importable. but python -m spacy info returns the same result.

**Reply by polm** [reply]
I see, in that case I suggest you follow @adrianeboyd's suggestions.

**Comment by adrianeboyd** [comment]
I suspect you've installed spacy in emulation mode and you're trying to run it in native mode, or vice versa.

It's also possible that `python` and `pip` don't point to the same installation? What do you get for `which python` and `which pip`?

You can see if one of these works:

```python
arch -x86_64 python -m spacy info
arch -arm64 python -m spacy info
```

It can be a bit difficult to keep python virtual environments straight given how seamlessly emulation runs in the background. I've found it easier to use a virtual environment with a python binary that **isn't** universal, so that you know that as long as you're in this venv, there's only one compatible platform.

I think the native miniforge OS X arm64 installer is a good choice if you're primarily using spacy (using binary conda-forge packages for spacy). Or if you're also using prodigy and need everything to run in x86_64 emulation mode, then miniconda x86_64 with python 3.8 would be a better option. It depends a bit on what you need to run, and these aren't the only options.
